### ISSUE TITLE
Nothing happens

### ISSUE THREAD

**Issue by knutanderss** [PaulBatchelor/worgle#3]
Just installed worgle and ran it on an .org file. 

Expected behavior: It produces a source file from the org-file (like emacs).
Actual behavior: Nothing happens. Return code is 0.

I've turned on warnings, but nothing is printed to the console.

**Comment by PaulBatchelor** [comment]
It doesn't terribly surprise me. Worgle only works on a *very* small subset of Org, which is good enough for my own personal needs. This whole codebase is written in a style of org that read by Worgle, so when in doubt look there. If you see it being used there, chances are it works.

Short of that, could you maybe upload a snippet of the org code you're trying to tangle?

**Comment by knutanderss** [comment]
I understand. Sorry, I can't upload since I don't own the rights to the file I'm working on.

**Comment by PaulBatchelor** [comment]
> Some sort of error message would be nice though.

Yeah, that's on me. Sorry about that. The parser isn't as robust as it could be. The parser is currently hand coded in ANSI C. I do this to make it very portable, but this also means things things move very slowly.

As I mentioned before, the parser is set to be a specific subset of org, but I don't explicitly say anywhere where what that subset actually *is*. Well, here's the gist of it (to be someday added to the README):

This is how a codeblock should look:

```org
#+NAME: file
#+BEGIN_SRC c :tangle main.c
#include <stdio.h>
int main(int argc, char *argv[])
{
    print("Hello World!\n");
    return 0;
}
#+END_SRC
```

- Every code block must have a `#+NAME:` tag that immediately precedes it
- Yes, this means the top-file too. In practice, I usually name it the filename. It seemss redundant, but it makes the implementation simpler.
- Spaces after `:` works. `#+NAME: foo` and `#+NAME:foo` are both valid
- `#+BEGIN_SRC` and `#+END_SRC` must be ALL CAPS. I went with this because it's the default behavor for the `<s` macro. It is not case insensitive
- indents won't work right now. this might be fixed in the future with #2 , but it isn't a high priority for me
- The language tag is essentially ignored. any language tag works
- The only tag that works is `:tangle`. Should only be used once per file. To append stuff across multiple files, make use of noweb-style named codeblocks.

Speaking of named codeblocks, here is how one can look:

```c
#+NAME: file
#+BEGIN_SRC c :tangle main.c
#include <stdio.h>
<<main_function>>
#+END_SRC

#+NAME: main_function
#+BEGIN_SRC c
int main(int argc, char *argv[])
{
    print("Hello World!\n");
    return 0;
}
#+END_SRC

```

Firstly, noweb expansion is always enabled (that's kind of the point). It doesn't need to be turned on because it can never be turned off.

Named blocks can be referenced before they are declared.

Preceding spaces are allowed so it matches local indentation rules  (no tab support at the moment). Worgle will look for matching `<<` and `>>` on the same line, and treat the stuff inside as a block name (surprisingly, this has never caused an issue for me). 

One named block per line, and it really should be just the block. Cool things like this WONT work:

```c
#+NAME: does_not_work
#+BEGIN_SRC c
if (<<condition>>) {
    do_something();
}
#+END_SRC
```

That's pretty much it, as far as what Worgle actually supports. It's very minimal, but it actually takes me pretty far.
